Source/UI/PaymentForm/ViewModel/DefaultPaymentViewModel.swift

```diff
@@ -175,6 +174,45 @@ extension DefaultPaymentViewModel: PaymentViewControllerDelegate {
 
   func cardholderIsUpdated(value: String) {
     cardDetails.name = value
+    validateMandatoryInputProvided()
+  }
+
+  private func validateMandatoryInputProvided() {
+    var isMandatoryInputProvided = false
+    defer { shouldEnablePayButton?(isMandatoryInputProvided) }
+
+    // If a scheme is not recorded the number hasn't started being inputted
+    // so we can safely know mandatory fields are not provided
+    guard let cardScheme = cardDetails.scheme else { return }
+
+    // Check if cardholder is required and if so whether it is provided
+    let isCardholderRequired = paymentFormStyle?.cardholderInput?.isMandatory == true
+    if isCardholderRequired && cardDetails.name.isEmpty { return }
```

Comment:
```suggestion
    guard paymentFormStyle?.cardholderInput?.isMandatory == false l, !cardDetails.name.isEmpty else { return }
```

Reply:
Your refactor suggestion doesn't do the same thing leaving early for nil. 

This is exactly why the original approach is trying to improve clarity by breaking complex logic down to readable logic.